girder/models/upload.py

```diff
@@ -179,6 +179,28 @@ def finalizeUpload(self, upload, assetstore=None):
 
         return file
 
+    def getTargetAssetstore(self, modelType, resource):
+        """
+        Get the assetstore for a particular target resource, i.e. where new
+        data within the resource should be stored. In girder core, this is
+        always just the current assetstore, but plugins may override this
+        behavior to allow for more granular assetstore selection.
+        """
+        eventParams = {'model': modelType, 'resource': resource}
+        event = events.trigger('model.upload.assetstore', eventParams)
+
+        if event.responses:
+            assetstore = event.responses[-1]
```

Comment:
If this isn't the only place we use the ``[-1]`` convention, then I suppose it's fine without a comment.